src/librustdoc/doctest/make.rs

```diff
@@ -458,9 +458,12 @@ fn parse_source(
 
     let sm = Arc::new(SourceMap::new(FilePathMapping::empty()));
     let translator = rustc_driver::default_translator();
-    info.supports_color =
-        HumanEmitter::new(stderr_destination(ColorConfig::Auto), translator.clone())
-            .supports_color();
+    let supports_color = match get_stderr_color_choice(ColorConfig::Auto, &std::io::stderr()) {
```

Comment:
I should've realized that much earlier: Obviously, like on stable, `rustdoc bad-doctest.rs --test | cat` / `rustdoc bad-doctest.rs --test > file.txt` still leads to colored output because libtest outputs this to stdout while we're querying stderr here.

I consider this a bug but I don't want it to be fixed in this PR since there are some subtleties. E.g., if you pass `--no-capture` (previously known as `--nocapture`) to rustdoc, then both output streams don't get captured meaning stderr would actually make sense. There are probably some other things that need consideration.